### ISSUE TITLE
dump the whole hls structure, not an mp4

### ISSUE THREAD

**Issue by zuzzurro** [osklil/hls-fetch#5]
How hard would it be to implement a recursive copy of the full hls structure instead of creating a single mp4 from a single bitrate?

**Comment by osklil** [comment]
That is basically what it does by default. There is no data modification, other than decryption using openssl. If you don't want decryption you can use --no-decrypt.

**Comment by osklil** [comment]
Do you have a case where hls-fetch does not work?

**Comment by zuzzurro** [comment]
I probably did not make myself clear. The tool works fine for me (thanks for it, it's been a life saver sometimes) when I need to dump a video.ts from an the net. But sometimes instead of the video I may want to duplicate the "folder" hierarchy locally so that I could serve the hls stream myself. This does not seem possible right now. If I remove the "-o" argument it still tried to dump into video.ts. I need to have all the metadata and all the chunks of all bitrates downloaded, with the same structure tha can be statically server with an http server instead.

**Comment by zuzzurro** [comment]
And, sorry, when I said writing a single mp4 I said something really stupid. What I meant of course was a single .ts file build from the video segments. mp4 is really the wrong thing to say in this context.

**Comment by osklil** [comment]
You mean save parts, along with m3us, into individual files? I don't know of any other hierarchy within the HLS technology.

**Comment by zuzzurro** [comment]
Yes. The hierarchy i'm talking about goes from the main m3u8 with multiple bitrates and audios, to the single bitrate manifest down to the segments. Sorry for not being more terminologically precise for the HLS platform...

**Comment by zuzzurro** [comment]
I found this script on github, that kind of does what I am looking for:

https://gist.githubusercontent.com/anonymous/4755742/raw/e7d209dfb68942f3a54140354657efce785049b2/hls-mirror.py

Just in case...

**Comment by osklil** [comment]
Good find! Yeah, it looks pretty easy. Just a matter of downloading and parsing the m3u8, and then download and save each separate piece. I guess it's something that could be implemented in hls-fetch as well...

**Comment by flavioribeiro** [comment]
there's also a project called [hlsclient](https://github.com/globocom/hlsclient) written in python for this purpose.
